src/issue.cpp

```diff
@@ -0,0 +1,346 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "libcellml/issue.h"
+#include "libcellml/types.h"
+#include "utilities.h"
+
+#include <map>
+#include <string>
+
+namespace libcellml {
+
+/**
+ * @brief The Issue::IssueImpl struct.
+ *
+ * The private implementation for the Issue class.
+ */
+struct Issue::IssueImpl
+{
+    std::string mDescription; /**< The string description for why this issue was raised. */
+    Issue::Cause mCause = Issue::Cause::UNDEFINED; /**< The Issue::Cause enum value for this issue. */
+    Issue::Level mLevel = Issue::Level::ERROR; /**< The Issue::Level enum value for this issue. */
```

Comment:
which I guess answers my question above. Need to check of the docs define this default...

Reply:
... which docs?

Reply:
I've added a section to the constructor for the `Issue` class.